src/shared-components/dropdown/index.tsx

```diff
@@ -6,27 +6,29 @@ import DesktopDropdown from './desktopDropdown';
 import allowNullPropType from '../../utils/allowNullPropType';
 
 export type OptionType = {
-  value: string;
+  value: string | undefined;
   label: string;
   disabled?: boolean;
 };
 
 type DropdownProps = {
-  value?: string;
+  borderRadius?: string;
   onChange: (option: OptionType) => void;
   options: OptionType[];
   optionsContainerMaxHeight?: string;
   textAlign?: 'left' | 'center';
+  value?: string;
 };
 
-const Dropdown = (props: DropdownProps) => {
```

Comment:
CHANGELOG:

1. Move `defaultProps` to ES6 defaults. 
    - Improved type-inference from relying on language built-ins without working out `defaultProps` limitations.
        - e.g., the compiler can infer that even though `borderRadius` is optional, since we're providing it with defaults when we pass it into `DesktopDropdown` and `MobileDropdown` we can type it there as required. 
    - There's still some inconsistencies: we're passing an ES6 default in `MobileDropdown` even though we typed it as a required prop. I'm not sure if that's just due to making things easier to test (related comment: https://github.com/curology/radiance-ui/pull/338/files#r488011625) but this PR already has enough ancillary changes so I didn't want to change too much. 
1. Modify type of `value` in `OptionType`
    - We rely on a falsy default for our dropdown: changing the type from `string` to `string | undefined` reflects this. 
    - There's some inconsistency in our falsy default--`null` vs. `undefined`. The type definition for `<option>` prefers `undefined` instead of `null` (`OptionHTMLAttributes<HTMLOptionElement>.value?: string | number | readonly string[] | undefined`) so this PR prefers that when making changes.